docs/5-stubbing-results.md

```diff
@@ -538,6 +538,83 @@ nextToken(3) // 'bar'
 nextToken(3) // 'foo'
 ```
 
+#### defer
+
+[Note: you probably don't need this option. Using it everywhere smells of overly
+defensive specifications.]
+
+By default, callback stubbings (whether configured via the `td.callback` matcher
+or by invoking `thenCallback`) are invoked synchronously. Since testdouble.js is
+designed for isolated unit tests, this is usually what you want, because
+comprehending and troubleshooting synchronous test scripts will always be much
+simpler than asynchronous ones. However, in the event that you want to ensure
+the subject isn't inadvertently relying on this synchronous execution of
+callbacks, you can ensure those callbacks are scheduled to a later execution
+stack by setting the `defer` option to `true`.
+
+Take this example of an erroneously passing test:
+
+```js
+// Subject under test
+function printBalance (id, fetchBalance, print) {
+  var balance;
+  fetchBalance(id, function (er, amount) {
+    balance = amount
+  })
+  print('Your balance is ' + balance)
+}
+
+// Test body
+var fetchBalance = td.function('.fetchBalance')
+var print = td.function('.print')
+td.when(fetchBalance(42)).thenCallback(null, 1337)
+
+printBalance(42, fetchBalance, print)
+
+td.verify(print('Your balance is 1337'))
+```
+
+The above passes, but suppose that in practice `fetchBalance` is going to
+invoke the callback asynchronously—if that's the case, then this passing test
+will be lying to us! To guard against this category of test smells, you can set
+the `defer` option when stubbing the async dependency, like so:
+
+```
+td.when(fetchBalance(42), {defer: true}).thenCallback(null, 1337)
+```
+
+Now the test above will fail—shiny! Keep in mind that you'll have to make the
+overall test asynchronous (e.g. a `done` callback in Mocha/Jasmine) when using
+the `defer` option.
```

Comment:
Of using a done callback in Mocha/Jasmine? The test of this feature uses one. I think people writing async tests will be familiar.